src/tokens/TokenImport.tsx

```diff
@@ -151,53 +164,70 @@ export default function TokenImportScreen(_: Props) {
     },
   })
 
-  const ensure0xPrefixOrEmpty = (address: string) =>
+  const isLoadingTokenDetails = validateContract.status === 'loading'
+
+  const ensure0xPrefixOrEmpty = (address: string): string =>
     !address || address.startsWith('0x') ? address : `0x${address}`
 
   const handleAddressBlur = async () => {
-    // ignore when handlePaste has already started validation (note - blur is called when focussed and keyboard is dismissed)
-    if (validateContract.status === 'loading' || error || tokenAddress.length === 0) return
-
+    if (!tokenAddress) return
     const addressWith0xPrefix = ensure0xPrefixOrEmpty(tokenAddress)
     setTokenAddress(addressWith0xPrefix)
-    if (validateAddress(addressWith0xPrefix)) {
-      // ignore propagated error as it's already handled, see https://github.com/slorber/react-async-hook/issues/85
-      await validateContract.execute(addressWith0xPrefix).catch(() => undefined)
-    }
   }
 
   const handlePaste = async (address: string) => {
     ValoraAnalytics.track(AssetsEvents.import_token_paste)
     const addressWith0xPrefix = ensure0xPrefixOrEmpty(address)
     setTokenAddress(addressWith0xPrefix)
     Keyboard.dismiss()
-    if (validateAddress(addressWith0xPrefix)) {
-      // ignore propagated error as it's already handled, see https://github.com/slorber/react-async-hook/issues/85
-      await validateContract.execute(addressWith0xPrefix).catch(() => undefined)
-    }
   }
 
+  useEffect(() => {
+    if (
+      walletAddress &&
+      tokenAddress &&
+      networkId &&
+      isAddress(walletAddress) &&
+      validateAddress(tokenAddress)
+    ) {
+      validateContract.execute(walletAddress, tokenAddress, networkId).catch(() => undefined)
+    }
+  }, [walletAddress, tokenAddress, networkId])
+
   const handleImportToken = () => {
     if (!tokenDetails) {
       Logger.warn(TAG, 'Token details not found, cannot import token')
       return
     }
 
-    const tokenId = getTokenId(networkId, tokenAddress.toLowerCase())
+    const networkId = tokenDetails.networkId
+    const tokenAddress = tokenDetails.address.toLowerCase()
+
+    // This should always be present, unless we support a network without a native token with icon,
+    // or if we can't fetch the token list (e.g. network issues?), should we fail instead to be safe?
+    const networkIconUrl = networkIconByNetworkId[networkId]
+
+    const tokenId = getTokenId(networkId, tokenAddress)
     ValoraAnalytics.track(AssetsEvents.import_token_submit, {
       tokenAddress,
       tokenSymbol: tokenDetails.symbol,
       networkId,
       tokenId,
+      networkIconUrl,
     })
 
     Logger.info(TAG, `Importing token: ${tokenId})})`)
-    dispatch(importToken({ ...tokenDetails, tokenId, networkId }))
+    dispatch(importToken({ ...tokenDetails, tokenId, networkIconUrl }))
```

Comment:
This will add the network icon statically, so if in the future we change the token icon this will be out of date.

So, probably we also want to update this in the the `fetchTokenBalances`. I'll add it in this PR, but then it got me thinking...
It's also possible that the `name`, `symbol` (and even the `decimal`) fields change for example if the token is implemented with a proxy. So, I wonder if we should also fetch those fields regularly to avoid showing old data.

I think it makes sense to update the network Icon, to avoid issues when updating icons, but updating the other fields sounds like overthinking.